moveit_py/src/moveit/moveit_core/robot_trajectory/robot_trajectory.cpp

```diff
@@ -57,6 +59,34 @@ set_robot_trajectory_msg(const std::shared_ptr<robot_trajectory::RobotTrajectory
   return robot_trajectory->setRobotTrajectoryMsg(robot_state, msg);
 }
 
+bool retimeTrajectory(std::shared_ptr<robot_trajectory::RobotTrajectory>& self, const std::string& algorithm,
+                      const double& velocity_scaling_factor, const double& acceleration_scaling_factor)
+{
+  bool success = false;
+  // Do the actual retiming
+  if (algorithm == "ruckig_smoothing")
+  {
+    trajectory_processing::RuckigSmoothing time_param;
+    success = time_param.applySmoothing(*self, velocity_scaling_factor, acceleration_scaling_factor, false, 0.01);
```

Comment:
`mitigate_overshoot` and `overshoot_threshold` are not configurable with this API